haystack/components/validators/json_schema.py

```diff
@@ -142,18 +143,22 @@ def run(
                 else:
                     validate(instance=content, schema=validation_schema)
 
-            return {"validated": messages}
+            return {"validated": [last_message]}
```

Comment:
Looks good but isn't this changing the method contract, do you remember why this change?

Reply:
My method contract you mean signature ?
The type is the same List[ChatMessage] but now this list is always of length 1.
The reason I did this is to not have exponentional cost by passing all message history to the LLM each time. Imagine your pipe does 20 loops, you used 20+19+18+17+16.... messages worth of tokens.
By passing only the last error (which should be the only needed to fix the json schema) you would only use 20 messages worth of tokens

Reply:
Ok I see; but one can also argue that previous failure attempts are useful as well. We should provide some top_k last messages argument, wdyt @lambda-science ?

Reply:
> Ok I see; but one can also argue that previous failure attempts are useful as well. We should provide some top_k last messages argument, wdyt @lambda-science ?

I'm sorry I completly missed to answer you and then went on holiday. I sounds like a good idea but might require a bit more coding ahah. Not sure if I have the space in the next few days for this (maybe during next week)